### PR TITLE
chore(main): Update “Plugin” Terminology to “Integration” for Consistency

### PR DESCRIPTION
**Pull request by JoeKarlsson**
#### Description:
This PR changes all instances of "plugin" to "integration" across the codebase, aligning the terminology with our hub and marketing website. This update includes:

- Replacing "plugin" with "integration" in code, comments, and documentation.
- Renaming files and directories where necessary to reflect the new terminology.
- Adding appropriate redirects for any renamed or relocated files to ensure no broken links.

If any instances were missed, I will open a follow-up PR to address them, but I believe I have covered most, if not all, of the occurrences.

#### Changes:
- Replaced "plugin" with "integration" throughout the project.
- Renamed files to match the updated terminology.
- Added redirects to maintain backward compatibility for renamed files.

#### Testing:
- Verified redirects work as expected.
- Tested links and file references to ensure no broken paths.

Let me know if you spot anything I might have missed!

image

### CODE REVIEW

**Comment by disq** [comment]
Old URLs should still work (redirect) to prevent SEO issues.

**Comment by JoeKarlsson** [comment]
> Old URLs should still work (redirect) to prevent SEO issues.

I think I'm confused, I added redirects for all URL changes. 🤔

**Comment by disq** [comment]
> > Old URLs should still work (redirect) to prevent SEO issues.
> 
> I think I'm confused, I added redirects for all URL changes. 🤔

Oh it's in [vercel.json](https://github.com/cloudquery/cloudquery/pull/19492/files#diff-21378a71578c160a16aa86a2ccc6120c09d1d10a3deaf16af651a228bd0cfde4) - ok, nevermind

File: contributing/creating_a_new_plugin.md [hunk 1]
```diff
@@ -1,17 +1,17 @@
-# Creating a New Plugin
```
Comment (hermanschaaf) [inline contributing/creating_a_new_plugin.md:1]: I think this file needs to be renamed

File: .github/ISSUE_TEMPLATE/new_destination_plugin.yml [hunk 2]
```diff
@@ -1,13 +1,13 @@
-name: 🔄 New Destination Plugin
```
Comment (hermanschaaf) [inline .github/ISSUE_TEMPLATE/new_destination_plugin.yml:1]: This file should also be renamed for consistency

File: .github/ISSUE_TEMPLATE/new_source_plugin.yml [hunk 3]
```diff
@@ -1,13 +1,13 @@
 name: 🔄 New Source Plugin
```
Comment (hermanschaaf) [inline .github/ISSUE_TEMPLATE/new_source_plugin.yml:1]: This file should also be renamed for consistency

File: contributing/creating_a_new_plugin.md (hunk 1 above)
Comment (JoeKarlsson) [reply contributing/creating_a_new_plugin.md:1]: Done!

File: .github/ISSUE_TEMPLATE/new_destination_plugin.yml (hunk 2 above)
Comment (JoeKarlsson) [reply .github/ISSUE_TEMPLATE/new_destination_plugin.yml:1]: Done!

File: .github/ISSUE_TEMPLATE/new_source_plugin.yml (hunk 3 above)
Comment (JoeKarlsson) [reply .github/ISSUE_TEMPLATE/new_source_plugin.yml]: Done!

**Comment by JoeKarlsson** [comment]
@erezrokah or @hermanschaaf @blesniewski - can I get a review?

File: website/pages/docs/advanced-topics/using-cloud-query-docker-registry-integrations-inside-a-containerized-environment.md [hunk 4]
```diff
@@ -29,12 +29,12 @@ spec:
   path: "cloudquery/postgresql"
   version: "VERSION_DESTINATION_POSTGRESQL"
 ```
-3. Run `cloudquery plugin install spec.yml` to pull the Docker plugin image locally.
+3. Run `cloudquery plugin install spec.yml` to pull the Docker integration image locally.
 
 ## Running a Sync
 
 1. [Create a CloudQuery API Key](https://docs.cloudquery.io/docs/deployment/generate-api-key) to be used with the Docker Compose file.
```
Comment (erezrokah) [inline website/pages/docs/advanced-topics/using-cloud-query-docker-registry-integrations-inside-a-containerized-environment.md:36]:
```suggestion
1. [Create a CloudQuery API Key](/docs/deployment/generate-api-key) to be used with the Docker Compose file.
```

Unrelated to your PR but this should be a relative link

**Review by erezrokah — APPROVED** [review]
Added a couple of comments, can be done in a follow up too.
We'll probably need to do a follow up to fix all the references to the old links in `vercel.json` to the new ones to avoid redirecting users every time

**Pull request merged**